src/policy/descriptor.rs

```diff
@@ -276,29 +277,50 @@ impl Descriptor<DescriptorPublicKey> {
 
 impl<Pk: MiniscriptKey> fmt::Display for Descriptor<Pk> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        fmt::Display::fmt(&self.policy, f)
+        write!(f, "sim({},{})", self.internal_key, self.policy)
     }
 }
 
 impl<Pk> FromStr for Descriptor<Pk>
 where
-    Pk: UnspendableKey + MiniscriptKey + FromStr,
+    Pk: MiniscriptKey + FromStr,
     Pk::Sha256: FromStr,
     <Pk as FromStr>::Err: ToString,
     <Pk::Sha256 as FromStr>::Err: ToString,
 {
-    type Err = miniscript::Error;
+    type Err = MSError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let policy = Policy::from_str(s)?;
-        Ok(Self::single_leaf(policy))
+        let (internal_key, policy) = s
+            .strip_prefix("sim(")
+            .and_then(|s| s.strip_suffix(')'))
+            .and_then(|s| {
+                let mut split = s.split(',');
+                let x = split.next()?;
+                let y = split.next()?;
```

Comment:
In afae99a2e01351a1b8d1eb7b93802e05b0501498:

This code will do the wrong thing if there is more than one `,`. You should use `splitn` rather than `split`.

Reply:
Fixed.